com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java

```diff
@@ -727,7 +728,13 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {
+        TerminalRule rule = (TerminalRule) grammarElement;
+        TerminalRule slCommentRule = formatter.getSLCommentRule();
+        TerminalRule mlCommentRule = formatter.getMLCommentRule();
+        return (rule.getName().equals(slCommentRule.getName()) && rule.isFragment() == slCommentRule.isFragment())

```

Comment:
what about writing the comparator once? I think the following code is nicer, what do you think?

      if (grammarElement instanceof TerminalRule) {
        TerminalRule rule = (TerminalRule) grammarElement;
        Comparator<TerminalRule> comparator = Comparator.comparing(TerminalRule::getName)//
            .thenComparing(TerminalRule::isFragment);
        return comparator.compare(formatter.getSLCommentRule(), rule) == 0 //
            || comparator.compare(formatter.getMLCommentRule(), rule) == 0;
      }

Reply:
ah this is nice. Im wondering tho, how deep should the comparison be?

Reply:
I am sorry, I do not know that either